request_handlers.h

```diff
@@ -1,19 +1,39 @@
-#ifndef REQUEST_HANDLER_H
-#define REQUEST_HANDLER_H
+#ifndef REQUEST_HANDLERS_H
+#define REQUEST_HANDLERS_H
 
 #include <boost/asio.hpp>
 
 using boost::asio::ip::tcp;
 
-namespace lightning_server {
-  // All request handlers return a pointer to the response buffer
-  // to be written back to the client.
-  namespace request_handlers {
-    // Echo the request that was received
-    void echoRequestHandler(char* request_buffer,
-                            char* response_buffer,
-                            size_t& response_buffer_size);
-  }
-}
+// This is an abstract base class
+class RequestHandler {
+  public:
+    RequestHandler() {};
+    virtual void handleRequest(const char* request_buffer,
+                               const size_t& request_buffer_size,
+                               char* &response_buffer,
+                               size_t& response_buffer_size) = 0;
+  private:
```

Comment:
Can you take out these `private` access-specifiers? We actually don't have any private methods or members at the moment. Plus, classes are private by default in C++.